scopes/workspace/install/install.main.runtime.ts

```diff
@@ -174,22 +163,79 @@ export class InstallMain {
       packageImportMethod: this.dependencyResolver.config.packageImportMethod,
       rootComponents: hasRootComponents,
     };
-    await installer.install(
-      this.workspace.path,
-      mergedRootPolicy,
-      compDirMap,
-      { installTeambitBit: false },
-      pmInstallOptions
-    );
+    // TODO: pass get install options
+    const installer = this.dependencyResolver.getInstaller({});
+    let current = await this._getComponentsManifests(installer, mergedRootPolicy, pmInstallOptions);
+    let prev: typeof current;
+    // TODO: this make duplicate
+    // this.logger.consoleSuccess();
+    // TODO: add the links results to the output
     await this.link({
-      linkTeambitBit: false,
+      linkTeambitBit: true,
       legacyLink: true,
       linkCoreAspects: this.dependencyResolver.linkCoreAspects(),
       linkDepsResolvedFromEnv: !hasRootComponents,
+      linkNestedDepsInNM: false,
+    });
+    const linkOpts = {
+      linkTeambitBit: false,
+      legacyLink: true,
+      linkCoreAspects: false,
+      linkDepsResolvedFromEnv: !hasRootComponents,
       linkNestedDepsInNM: !this.workspace.isLegacy && !hasRootComponents,
+    };
+    /* eslint-disable no-await-in-loop */
+    do {
+      this.workspace.consumer.componentLoader.clearComponentsCache();
+      this.workspace.clearCache();
```

Comment:
I wonder if we can improve the cache cleaning here somehow.
as this is a very expensive op (not the clean, the reload again once it's cleaned)
maybe we should somehow check what should be cleaned by checking if the component's env wasn't loaded in the previous iteration or something like this.
we can keep this optimization to a different PR, but may be worth discussing it also with @davidfirst

Reply:
actually, it seems to work without clearing the cache. So I'll remove these lines